app/controllers/rankings_controller.rb

```diff
@@ -1,79 +1,123 @@
+      flash[:message] = "Access denied."
+    end
+  end
+
+  def end
+
+  end
+
+  # POST /rankings
+  # POST /rankings.json
+  def create
+    # @ranking = Ranking.new(ranking_params)
+    #
+    # respond_to do |format|
+    #   if @ranking.save
+    #     format.html { redirect_to @ranking, notice: 'Ranking was successfully created.' }
+    #     format.json { render :show, status: :created, location: @ranking }
+    #   else
+    #     format.html { render :new }
+    #     format.json { render json: @ranking.errors, status: :unprocessable_entity }
+    #   end
+    # end
+
+    @times.ids.each do |id|
+      unless params.keys.include? id.to_s
+        flash[:error] = "Error: please fill in an option for each time."
+        redirect_to edit_project_participant_ranking_path(:secret_id => @participant.secret_id) and return
+      end
+    end
+
+    @times.ids.each do |id|
+      rank_num = params[id.to_s].to_i
+      new_rank = Ranking.new(:id => @ranking.id, :rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      existing_rank = Ranking.find_by(:id => @ranking.id, :project_time_id => id)
+
+      if existing_rank
+        existing_rank.destroy
+      end
+      new_rank.save!
+    end
+
+    redirect_to end_project_participant_ranking_path
+  end
+
+  # PATCH/PUT /rankings/1
+  # PATCH/PUT /rankings/1.json
+  def update
+    @times.ids.each do |id|
+      unless params.keys.include? id.to_s
+        flash[:error] = "Error: please fill in an option for each time."
+        redirect_to edit_project_participant_ranking_path(:secret_id => @participant.secret_id) and return
+      end
+    end
+
+    @times.ids.each do |id|
+      rank_num = params[id.to_s].to_i
+      new_ranking = Ranking.new(:rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      existing_ranking = Ranking.find_by(:participant_id => @participant.id, :project_time_id => id)
+
+      if existing_ranking
+        Ranking.find_by(:participant_id => @participant.id, :project_time_id => id).update(
+            :rank => rank_num, :participant_id => @participant.id, :project_time_id => id)
+      else
+        new_ranking.save!
+      end
+    end
+
+    redirect_to end_project_participant_ranking_path
+  end
+
+  # DELETE /rankings/1
+  # DELETE /rankings/1.json
+  # def destroy
+  #   @ranking.destroy
+  #   respond_to do |format|
+  #     format.html { redirect_to rankings_url, notice: 'Ranking was successfully destroyed.' }
+  #     format.json { head :no_content }
+  #   end
+  # end
+
   private
     # Use callbacks to share common setup or constraints between actions.
-    #def set_ranking
-     # @ranking = Ranking.find(params[:participant_id])
-    #end
+    def set_fields
+      @project = Project.find(params[:project_id])
+      @participant = Participant.find_by(:project_id => params[:project_id], :id => params[:participant_id])
+      @times = @project.project_times
+      @rankings = Ranking.all
```

Comment:
should do `@participant.rankings` (participant has_many rankings)
you're fetching all the rankings for the entire database here, just need it for this participant right?